### PR TITLE
Allow CRLF at the start of multiline strings

### LINKED ISSUE #254: Multi-line strings don't work on Windows
<details><summary>Original error</summary>
<p>

I recently tried to use a new version of Gren (previously on 0.2.0) and was unuable to get it to compile. As a result I tried just doing a simple project with default dependencies.

My `gren.json` (created by gren init) is below:

```
{
    "type": "application",
    "platform": "browser",
    "source-directories": [
        "src"
    ],
    "gren-version": "0.4.4",
    "dependencies": {
        "direct": {
            "gren-lang/browser": "4.0.0",
            "gren-lang/core": "5.0.2"
        },
        "indirect": {
            "gren-lang/url": "4.0.0"
        }
    }
}
```

and it results in this error:

```
Dependency problem!
-- PROBLEM BUILDING DEPENDENCIES -----------------------------------------------

I ran into a compilation error when trying to build the following package:

    gren-lang/browser 4.0.0

This probably means it has package constraints that are too wide. It may be
possible to tweak your gren.json to avoid the root problem as a stopgap. Head
over to https://gren-lang.org/community to get help figuring out how to take
this path!

Note: To help with the root problem, please report this to the package author
along with the following information:

    gren-lang/core 5.0.2
    gren-lang/url 4.0.0

If you want to help out even more, try building the package locally. That should
give you much more specific information about why this package is failing to
build, which will in turn make it easier for the package author to fix it!
```

</p>
</details> 

Actually it looks like the error is that Git on Windows likes changing line endings to \r\n rather than just \n and Gren doesn't like that with multiline strings:

```
AppData\Local\gren\0.4.4\packages\gren-lang\browser\4.0.0
> gren make
Dependencies ready!
Detected problems in 2 modules.
-- MULTILINE STRING WITHOUT LEADING NEWLINE ---------- src\Debugger\History.gren

The contents of a multiline sting must start on a new line

447|     Html.node "style" [] [ text """
                                     ^
Add a """ a new line right after the opening quotes.

Note: Here is a valid multi-line string for reference:

    """
    # Multi-line Strings

    - start with triple double quotes
    - write whatever you want
    - no need to escape newlines or double quotes
    - end with triple double quotes
    """

                                                         Debugger.History  Γåæ
====o======================================================================o====
    Γåô  Debugger.Overlay


-- MULTILINE STRING WITHOUT LEADING NEWLINE ---------- src\Debugger\Overlay.gren

The contents of a multiline sting must start on a new line

297|     """
         ^
Add a """ a new line right after the opening quotes.

Note: Here is a valid multi-line string for reference:

    """
    # Multi-line Strings

    - start with triple double quotes
    - write whatever you want
    - no need to escape newlines or double quotes
    - end with triple double quotes
    """
```

My system git has `core.autocrlf` set to true, but everything else is generally happy.

### PR DESCRIPTION
**Pull request by mbartlett21**
This should fix #254.

I also fixed another bug while I was at it. Gren was silently dropping the last newline if the string didn't end with a newline.

### CODE REVIEW

File: compiler/src/Parse/String.hs [hunk 1]
```diff
@@ -175,7 +175,7 @@ multiString pos end row _ _ sr sc =
     then Err sr sc E.StringEndless_Multi
     else
       let !word = P.unsafeIndex pos
-       in if word == 0x0A {- \n -}
+       in if word == 0x0A || word == 0x0D {- \n or \r -}
```
Comment (robinheghan) [inline compiler/src/Parse/String.hs:178]: This should probably be rewritten to check for the existance of `\r\n` and not just `\r`. You should probably also make a similar change to `multiStringBody` if you want this sequence to be valid within multistrings as well.

File: compiler/src/Reporting/Error/Syntax.hs [hunk 2]
```diff
@@ -2980,7 +2980,7 @@ toStringReport source string row col =
               source
               region
               Nothing
-              ( D.reflow "The contents of a multiline sting must start on a new line",
+              ( D.reflow "The contents of a multiline string must start on a new line",
```
Comment (robinheghan) [inline compiler/src/Reporting/Error/Syntax.hs:2983]: Good catch!

File: tests/Parse/MultilineStringSpec.hs [hunk 3]
```diff
@@ -21,6 +21,11 @@ spec = do
         "normal string"
         "\"\"\"\nnormal string\"\"\""
 
+    it "crlf regression test" $ do
+      parse
+        "normal string"
+        "\"\"\"\r\nnormal string\"\"\""
```
Comment (robinheghan) [inline tests/Parse/MultilineStringSpec.hs:27]: You should probably add the following tests

* A multiline string with a leading `\r` (missing `\n`) shouldn't be allowed
* `\r\n` should be allowed in the middle of a multi-string as well

**Review by robinheghan — CHANGES_REQUESTED** [review]
Good first attempt, but I think we can do better.

Also, please add your name to the CONTRIBUTORS file, if it isn't there already.

**Comment by robinheghan** [comment]
Code formatting check fails.

To apply correct code formatting, install `ormolu` and run `ormolu --mode inplace $(git ls-files '*.hs')`

**Comment by mbartlett21** [comment]
@robinheghan 
I've added another test for if it doesn't end with a newline, which was previously wrong.
I.e.
```
"""testing
testing"""
```
previously dropped the newline

**Review by robinheghan — APPROVED** [review]
Thank you for this <3

**Pull request merged**